subprojects/core/src/integTest/groovy/org/gradle/api/internal/cache/CacheConfigurationsIntegrationTest.groovy

```diff
@@ -77,20 +120,69 @@ class CacheConfigurationsIntegrationTest extends AbstractIntegrationSpec {
                 settings.caches {
                     markingStrategy = MarkingStrategy.NONE
                     cleanup = Cleanup.DISABLED
-                    releasedWrappers.removeUnusedEntriesOlderThan = ${releasedDistTimestamp}
-                    snapshotWrappers.removeUnusedEntriesOlderThan = ${snapshotDistTimestamp}
-                    downloadedResources.removeUnusedEntriesOlderThan = ${downloadedResourcesTimestamp}
-                    createdResources.removeUnusedEntriesOlderThan = ${createdResourcesTimestamp}
+                    releasedWrappers.entryRetentionAge = ${releasedDistTimestamp}L
+                    snapshotWrappers.entryRetentionAge = ${snapshotDistTimestamp}L
+                    downloadedResources.entryRetentionAge = ${downloadedResourcesTimestamp}L
+                    createdResources.entryRetentionAge = ${createdResourcesTimestamp}L
+                    buildCache.entryRetentionAge = ${buildCacheResourcesTimestamp}L
```

Comment:
🤔 I wonder if we should even have this test given this is private api.  I think the test was around previously because it was overlooked when we pulled this out of the public api and the magic of groovy just allowed it to continue working.  I suppose it's fine if you want to keep it, but it might make sense to remove it.

Reply:
I've removed any integration tests that directly set the `Provder` value, but added/modified one to test the new `removeEntriesUnusedSince` method: while this is internal API (currently), it is clearly user-facing.

We should probably just promote it to public API as `@Incubating`.